plugins/usertiming.js

```diff
@@ -0,0 +1,110 @@
+/**
+\file usertiming.js
+Plugin to collect metrics from the W3C User Timing API.
+For more information about User Timing,
+see: http://www.w3.org/TR/user-timing/
+
+This plugin is dependent on the UserTimingCompression library
+see: https://github.com/nicjansma/usertiming-compression.js
+UserTimingCompression must be loaded before this plugin's init is called.
+*/
+
+(function() {
+
+	BOOMR = BOOMR || {};
+	BOOMR.plugins = BOOMR.plugins || {};
+
+	if (BOOMR.plugins.UserTiming) {
+		return;
+	}
+
+	/**
+	 * @returns String compressed user timing data that occurred since the last call
+	 */
+	function getUserTiming() {
+		var map, res, now = BOOMR.now();
+
+		map = UserTimingCompression.getCompressedUserTiming(impl.options);
+		res = UserTimingCompression.compressForUri(map);
+		impl.options["from"] = now;
```

Comment:
I'm still concerned about this resulting in us missing data.  The use case is as follows:

   1. Page calls `performance.mark("mark-xyz");` at `time=t1`
   2. Some event causes beacon to be sent at `time=t2`
   3. Page calls `performance.measure("measure-abc", "mark-xyz");` at `time=t3`
   4. Some other event causes a beacon to be sent at `time=t4`

In this case, the first beacon will include `mark-xyz` (`t1`), and set `from` to `t2`.
The second beacon will include nothing because `measure-abc` has a start time of `t1`, but `from` is set to `t2` which is `> t1`

I don't know of a good way to solve this here, so perhaps this actually needs to be fixed in `UserTimingCompression`.

@nicjansma can you comment on this scenario?